lib/ansible/utils/collection_loader/_collection_finder.py

```diff
@@ -277,13 +291,31 @@ def find_module(self, fullname, path=None):
                         # might not be in some other situation...
                         return None
 
-                spec = self._file_finder.find_spec(fullname)
-                if not spec:
-                    return None
-                return spec.loader
+                return self._file_finder
+
+            # call py2's internal loader
+            return pkgutil.ImpImporter(self._pathctx)
+
+    def find_module(self, fullname, path=None):
+        # we ignore the passed in path here- use what we got from the path hook init
+        finder = self._get_finder(fullname)
+        if finder is not None:
+            return finder.find_module(fullname, path=[self._pathctx])
+        else:
+            return None
+
+    def find_spec(self, fullname, target=None):
+        split_name = fullname.split('.')
+        toplevel_pkg = split_name[0]
+
+        finder = self._get_finder(fullname)
+        if finder is not None:
+            if toplevel_pkg == 'ansible_collections':
+                return finder.find_spec(fullname, path=[self._pathctx])
             else:
-                # call py2's internal loader
-                return pkgutil.ImpImporter(self._pathctx).find_module(fullname)
+                return finder.find_spec(fullname)
+        else:
+            return None
```

Comment:
nitpick: I think this could have less nesting.
```suggestion
        if finder is None:
            return None

        if toplevel_pkg == 'ansible_collections':
            return finder.find_spec(fullname, path=[self._pathctx])

        return finder.find_spec(fullname)
```